### PR TITLE
New Mountain Puzzles, Fix Scene Transition Softlock

### PR DESCRIPTION
**Pull request by ChaseOB**
## Description

New Mountain Puzzles
Tile 1: Added empty bucket next to NPC. This can be filled with lava from tile 2. Doing so will make NPC give you basement key, which is now needed to enter basement on tile 2

Tile 6: Added sad elevator man NPC. When you send a minecart down the elevator, he will enable the pipe item to fix the heater

New classes:
-LavaBucket
-LavaFallBottom

Changed Classes:
-MinecartElevator: tracks if a minecart has gone down
-PlayerInventory: added remove item method w/o return for event invoke

Other Changes:
-changes in Mountain scene
-added sprite for bucket

Minecart Save/Load and UI Tracker
Minecart UI Tracker now updates based on minecart contents. Minecart contents are saved so that they can be restored on load. Minecart position is not saved, it is always returned to the station

Changed Classes:
-Minecart: added UI trackers, save/load
-MountainCaveWall: added null check for loading in caves scene, seperate spawn for saved MC load

Fix Scene Transition Softlock
You can no longer open the artifact when transitioning between scenes. Game no longer softlocks if you attempt to do so.

Changed Classes:
-moved fade from black call from individual SGrids to main SGrid
-SceneChanger: force UI close before fade to black instead of after
-UIManager and UIArtfactMenus: more control over setting canOpenMenus

## Related Task

https://trello.com/c/CQN5x909/318-gdc-igf-bugs
https://trello.com/c/yoMxVeuA/288-mountain-puzzle-changes
https://trello.com/c/SeqLv2xG/263-better-minecart

## How Has This Been Tested?

-Played through mountain
-Spammed tab (shift) while transitioning, game did not softlock

### CODE REVIEW

File: Slider/Assets/Scripts/Map/Mountain/Minecart/Minecart.cs [hunk 1]
```diff
@@ -149,6 +152,19 @@ public override STile DropItem(Vector3 dropLocation, System.Action callback=null
 
     #endregion
 
+
+    private void AddTracker()
```
Comment (randomerz) [inline Slider/Assets/Scripts/Map/Mountain/Minecart/Minecart.cs:156]: this should prob just be a `GetTrackerSprite()` and then add the tracker in a single line

File: Slider/Assets/Scripts/Map/Mountain/Minecart/Minecart.cs [hunk 2]
```diff
@@ -459,12 +475,19 @@ private static Vector3 getDirectionAsVector(int dir)
             mcState = MinecartState.RepairParts;
```
Comment (randomerz) [inline Slider/Assets/Scripts/Map/Mountain/Minecart/Minecart.cs:475]: do UnityEvents not allow enums to be passed in for `UpdateState()`

File: Slider/Assets/Scripts/Map/Mountain/Minecart/MinecartElevator.cs [hunk 3]
```diff
@@ -36,10 +37,12 @@ public void FixElevator()
 
     public void SendMinecartDown(Minecart mc)
     {
-        if(!isFixed || !isOpen)
+        if(!isFixed)
             return;
         mc.StopMoving();
         StartCoroutine(WaitThenSend(mc, bottomPosition.transform.position, 3));
+        hasGoneDown = true;
+        Debug.Log("send down");
```
Comment (randomerz) [inline Slider/Assets/Scripts/Map/Mountain/Minecart/MinecartElevator.cs:45]: begone, debug

File: Slider/Assets/Scripts/Player/PlayerInventory.cs [hunk 4]
```diff
@@ -131,6 +131,11 @@ public static void NextItem()
         }
     }
 
+    public static void RemoveItemNoReturn()
```
Comment (randomerz) [inline Slider/Assets/Scripts/Player/PlayerInventory.cs:134]: leave a comment that this is exposed for editor prob

**Review by randomerz — APPROVED** [review]
looks good, tysm for fixing all the bugs \o/

File: Slider/Assets/Scripts/Map/Mountain/Minecart/Minecart.cs (hunk 2 above)
Comment (ChaseOB) [reply Slider/Assets/Scripts/Map/Mountain/Minecart/Minecart.cs:475]: Nope, they do not!

**Pull request merged**